library/Icingadb/Util/PerfData.php

```diff
@@ -426,12 +426,15 @@ protected function parse()
         $parts = explode(';', $this->perfdataValue);
 
         $matches = array();
-        if (preg_match('@^(-?(?:\d+)?(?:\.\d+)?)([a-zA-Z%°]{1,3})$@u', $parts[0], $matches)) {
+        if (preg_match('@^((?:U|(?:-?\d+)?(?:\.\d+)?))([a-zA-TV-Z%°μ]{1,3})$@u', $parts[0], $matches)) {
```

Comment:
The `-?` can't be moved inside the first non-capture group. `.9kW` is a valid value. See https://github.com/Icinga/icingaweb2/issues/4690 for details.

What's `A-TV-Z` for? The regex currently doesn't match `'U'` anyway. And even if it does, the `U` is consumed early and the range can't consume it again.

The `μ` is not necessary. You already mentioned Icinga's normalization, so why do you think it is?

Please change the regex so that the following matches: `-.9kW`, `UkW`, `-1.9kW`, `.9kW`, `1.9kW`

Reply:
`U|` is not optional. In my tests this always consumes a leading `U`, no matter what follows. `-kW` isn't valid anyway and just `kW` as well. (makes no sense) And even if supplied this way, it's correctly identified as value (`-`) and uom (`kW`).

So I'm fine with the usage of `is_numeric` to decide whether to null the value again. But the `A-Z` range can be kept in combination with `U|` at the front.

I've also optimized it a bit: `@^(U|-?(?:\d+)?(?:\.\d+)?)([a-zA-Z%°]{1,3})$@u`

I've run unittests against this and I don't get any (new) errors. Or have you an example that doesn't work with this?